daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -22,11 +22,14 @@ import org.slf4j.LoggerFactory
 import scala.collection.JavaConverters._
 import scala.util.control.NoStackTrace
 
-object Speedy {
+private[lf] object Speedy {
+
+  // fake participant to generate a new transactionSeed when running scenarios
+  private val scenarioServiceParticipant = Ref.ParticipantId.assertFromString("scenario-service")
 
   // Would like these to have zero cost when not enabled. Better still, to be switchable at runtime.
-  val enableInstrumentation: Boolean = false
-  val enableLightweightStepTracing: Boolean = false
+  private val enableInstrumentation: Boolean = false
```

Comment:
Can this be even more strict?
```suggestion
  private[this] val enableInstrumentation: Boolean = false
```

Same for other `private` uses below.

Reply:
:+1: